gtsam/slam/OrientedPlane3Factor.h

```diff
@@ -48,10 +49,27 @@ class OrientedPlane3Factor: public NoiseModelFactor2<Pose3, OrientedPlane3> {
   virtual Vector evaluateError(const Pose3& pose, const OrientedPlane3& plane,
       boost::optional<Matrix&> H1 = boost::none, boost::optional<Matrix&> H2 =
           boost::none) const {
-    OrientedPlane3 predicted_plane = OrientedPlane3::Transform(plane, pose, H1,
-        H2);
     Vector err(3);
-    err << predicted_plane.error(measured_p_);
+
+    if (H1 || H2) {
+      Matrix H1_1, H2_1;
+      OrientedPlane3 predicted_plane = OrientedPlane3::Transform(plane, pose, H1_1, H2_1);
+      err << predicted_plane.error(measured_p_);
+      // Numerically calculate the derivative since this function doesn't provide one.
+      auto f = boost::bind(&OrientedPlane3::Error, _1, _2);
```

Comment:
move this logic into OrientedPlane.cpp, to calculate the optional derivatives of the error method, then call

```
Matrix33 error_H_predicted, error_H_measured;
err << predicted.error(measured_p_, error_H_predicted, error_H_measured);
```